lambda/handler.go

```diff
@@ -99,6 +99,31 @@ func WithEnableSIGTERM(callbacks ...func()) Option {
 	})
 }
 
+// ValidateHandlerFunc validates the handler against the criteria for Start and returns an error
+// if the criteria are not met
+func ValidateHandlerFunc(handlerFunc interface{}) error {
```

Comment:
I don't think I want to include this function - if you want to continue the discussion on it, then please move it to a separate PR so that I can merge in the new `StartHandlerFunc`

Reply:
Fair enough!